### PR TITLE
Fix #67 - Raise ImproperlyConfigured from Setting.check

### LINKED ISSUE #67: Raise ImproperlyConfigured on invalid setting
It seems, appsettings produce various exceptions on errors. To denote errors in settings correctly the `ImproperlyConfigured` should be raised instead.

### PR DESCRIPTION
**Pull request by ziima**

### CODE REVIEW

File: src/appsettings/settings.py [hunk 1]
```diff
@@ -132,6 +133,9 @@ def value(self):
 
         Returns:
             object: the transformed raw value.
+
+        Raises:
+            ImproperlyConfigured: The setting doesn't have a valid valid.
```
Comment (pawamoy) [inline src/appsettings/settings.py]: valid value :D

File: src/appsettings/settings.py [hunk 2]
```diff
@@ -145,6 +149,9 @@ def get_value(self):
...
```
Comment (pawamoy) [inline src/appsettings/settings.py]: valid value again haha. I'll stop on this one ^^

File: src/appsettings/settings.py (hunk 1 above)
Comment (ziima) [reply src/appsettings/settings.py]: Oops :laughing:

**Comment by ziima** [comment]
Fixed. I also deleted a note in README, which is no longer valid.

File: src/appsettings/settings.py [hunk 3]
```diff
@@ -184,8 +191,7 @@ def check(self):
         Validate the setting value.
 
         Raises:
-            AttributeError: if the setting is missing and required.
-            ValueError: if the raw value is invalid.
+            ImproperlyConfigured: The setting doesn't have a valid valid.
```
Comment (pawamoy) [inline src/appsettings/settings.py]: You forgot one :scream: I thought you'd do a find-replace haha

Comment (ziima) [reply src/appsettings/settings.py]: I did, but I probably shut it down too early :) Anyway, it's fixed.

**Pull request merged**